framework/src/userobjects/PropertyReadFile.C

```diff
@@ -50,14 +53,25 @@ PropertyReadFile::validParams()
       "Periodic or non-periodic grain distribution: Default is non-periodic");
   params.addParam<bool>(
       "use_zero_based_block_indexing", true, "Are the blocks numbered starting at zero?");
+
+  // Set an execution schedule to what makes sense currently
+  // We do not allow INITIAL because we read the file at construction
+  ExecFlagEnum & exec_enum = params.set<ExecFlagEnum>("execute_on", true);
+  exec_enum.removeAvailableFlags(EXEC_INITIAL, EXEC_FINAL, EXEC_LINEAR, EXEC_NONLINEAR);
+  params.setDocString("execute_on", exec_enum.getDocString());
```

Comment:
After this change, some users are getting the following error message:
```
Invalid option "INITIAL" in MultiMooseEnum.  Valid options (not case-sensitive) are "NONE POSTCHECK TIMESTEP_END TIMESTEP_BEGIN MULTIAPP_FIXED_POINT_END MULTIAPP_FIXED_POINT_BEGIN CUSTOM"
```

I understand the logic behind disabling INITIAL, FINAL, LINEAR, NONLINEAR because file read is performed at construction time. My question is shouldn't we just disable the `execute_on` parameter entirely?

Reply:
maybe tag @GiudGiud

Reply:
The first file is read on construction. 

The other files can be read based on the execution schedule which is set by the execute_on flag